File: Soutenance 2 End/Miny-RPG Editor/Cells.vb
Public Class Cells

    Public Shared Function GetId(ByVal x As Integer, ByVal y As Integer) As Integer
        Return y * MAP_WIDTH + x
    End Function

    Public Shared Function GetPos(ByVal CellId As Integer) As Point
        Return New Point(CellId - Math.Floor(CellId / MAP_WIDTH) * MAP_WIDTH, Math.Floor(CellId / MAP_WIDTH))
    End Function

    Public Shared Function GetScreenPos(ByVal CellId As Integer) As PointF
        Dim p As Point = GetPos(CellId)
        Return New PointF(p.X * TILE_WIDTH, p.Y * TILE_WIDTH)
    End Function

End Class

File: Soutenance 3/MapEditorReborn/MapEditorReborn/Cells.cs
using System;
using System.Drawing;

namespace MapEditorReborn
{
    public class Cells
    {
        public static Point DeltaScrolling { get; set; }

        public static int GetId(int x, int y)
        {
            return y * Constants.MapWidth + x;
        }

        public static Point GetPos(int cellId)
        {
            return new Point(cellId - (int)(Math.Floor((double)cellId / Constants.MapWidth)) * Constants.MapWidth, (int)(Math.Floor((double)cellId / Constants.MapWidth)));
        }

        public static PointF GetScreenPos(int cellId)
        {
            Point p = GetPos(cellId);
            p = new Point(p.X + DeltaScrolling.X, p.Y + DeltaScrolling.Y);
            return new PointF(p.X * Constants.TileWidth, p.Y * Constants.TileWidth);
        }
    }
}
